score-lang-compiler/src/main/java/org/openscore/lang/compiler/modeller/transformers/AsyncLoopForTransformer.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * (c) Copyright 2014 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *******************************************************************************/
+package org.openscore.lang.compiler.modeller.transformers;
+
+import org.apache.commons.lang.StringUtils;
+import org.openscore.lang.entities.AsyncLoopStatement;
+import org.openscore.lang.entities.ScoreLangConstants;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Date: 3/17/2015
+ *
+ * @author Bonczidai Levente
+ */
+@Component
+public class AsyncLoopForTransformer implements Transformer<String, AsyncLoopStatement> {
```

Comment:
Hi,

to avoid code duplication, introduced the following abstractions:

- LoopStatement
    - AsyncLoopStatement
    - ForLoopStatement
        - ListForLoopStatement
        - MapForLoopStatement
        
and

- AbstractForTransformer
    - ForTransformer
    - AsyncLoopForTransformer

please check and let me know WDYT